### PR TITLE
Add Segment Anything model + new utils

### PR DESCRIPTION
**Pull request by jan-grzybek-ampere**
* SAM network
* move CI test to Ubuntu 22.04
* helper func: wrapper for argparse
* add tqdm bar to runs with --timeout
* some changes to COCO class to support segmentation task
* new PyTorchRunnerV2 class -> simplifies what PyTorchRunner did and shifts the responsibility for torch compilation of the network to the user of the class (i.e. torch compilation of network's graph should take place in runner script of given network, instead of leading to ever increasing complexity of PyTorchRunner's __init__)

### CODE REVIEW

**Comment by MarcelWilnicki** [comment]
where can I find the model?

**Comment by MarcelWilnicki** [comment]
consider splitting it into individual PRs. PR per feature. This will make reviewing easier.

**Comment by jan-grzybek-ampere** [comment]
> where can I find the model?

There is a read-me covering that

**Comment by jan-grzybek-ampere** [comment]
@MarcelWilnicki do you have any comments aside from <python3.9 incompatibility?

File: utils/pytorch.py [hunk 1]
```diff
@@ -123,19 +121,95 @@ def print_performance_metrics(self, batch_size):
...
-                writer.writerow(self.__finish_times[2:])
+        bench_utils.dump_csv_results(batch_size, self.__start_times, self.__finish_times)
 
         if self.__is_profiling:
             print(self.__profile.key_averages().table(sort_by='cpu_time_total', row_limit=50))
             torch._C._aio_profiler_print()
         return perf
 
+
+class PyTorchRunnerV2:
```
Comment (dkupnicki) [inline utils/pytorch.py:132]: It's not that much different from V1. What's the plan here? Do we want to eventually move all of the models to use V2 and it's just a temporary measure to have both of them at the same time?

**Review by dkupnicki — CHANGES_REQUESTED** [review]
Good PR, just a few small things

File: utils/helpers.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import argparse
+
+
+SUPPORTED_FRAMEWORKS = ["tf", "ort", "pytorch", "ctranslate2", "tflite"]
+
+
+class DefaultArgParser:
+    def __init__(self, supported_frameworks: list[str]):
```
[the remark this answers is not in the document]
Comment (smamindl) [reply utils/helpers.py:8]: I see the same issue - TypeError: 'type' object is not subscriptable

**Comment by jan-grzybek-ampere** [comment]
@dkupnicki @MarcelWilnicki @smamindl I've addressed the issues, please have another look

**Review by dkupnicki — APPROVED** [review]
Looks good 👍

**Pull request merged**